### PR TITLE
Docs: Edit pxf filter pushdown docs

### PR DESCRIPTION
**Pull request by dyozie**

### CODE REVIEW

**Comment by dyozie** [comment]
@leskin-in - Added further edits to your work here.  Will also use this to confirm Hive/HBase pushdown support from @kongyew

**Comment by leskin-in** [comment]
Well done, @dyozie! To me, everything looks fine now.

File: gpdb-doc/markdown/pxf/using_pxf.html.md.erb [hunk 1]
```diff
@@ -88,6 +88,37 @@ To write data to an external data store with PXF, you create an external table w
...
+
+You enable or disable filter pushdown for all external data accessors in Greenplum Database by setting the `gp_external_enable_filter_pushdown` configuration parameter. The default value is `off`; set it to `on` to enable filter pushdown.
+
+**Note:** Some data sources do not support filter pushdown. Also, filter pushdown may not be supported with certain data types or operators. If a query accesses a data source that does not support filter push-down for the query constraints, the query is instead executed without filter pushdown (the data is first transferred to Greenplum Database and then filtered). 
+
+PXF accesses data sources using different connectors, and filter pushdown support is determined by the specific connector implementation. The following connectors support filter pushdown:
+* HBase
+* Hive
+
+PXF filter pushdown is supported for these data types:
```
Comment (lisakowen) [inline gpdb-doc/markdown/pxf/using_pxf.html.md.erb]: i am wondering how the data types and operators listed here relate to those that were specifically called out for the Hive ORC* profiles in the HAWQ docs (http://hawq.incubator.apache.org/docs/userguide/2.3.0.0-incubating/pxf/HivePXF.html#hiveorc-intro)?  i.e. are those identified for the ORC profiles possibly a subset?

File: gpdb-doc/markdown/pxf/using_pxf.html.md.erb [hunk 2]
```diff
@@ -88,6 +88,37 @@ To write data to an external data store with PXF, you create an external table w
 GRANT INSERT ON PROTOCOL pxf TO bill; 
 ```
 
+## <a id="filter-pushdown"></a>Configuring Filter Pushdown
+
+PXF supports filter pushdown. When filter pushdown is enabled, the constraints from the `WHERE` clause of a `SELECT` query can be extracted and passed to the external data source for filtering. This process can improve query performance, and can also reduce the amount of data that is transferred to Greenplum Database.
+
+You enable or disable filter pushdown for all external data accessors in Greenplum Database by setting the `gp_external_enable_filter_pushdown` configuration parameter. The default value is `off`; set it to `on` to enable filter pushdown.
```
[the remark this answers is not in the document]
Comment (dyozie) [reply gpdb-doc/markdown/pxf/using_pxf.html.md.erb]: I think a procedure might be in order at some point, but probably not now given that we're not officially supporting pushdown in this release.

File: gpdb-doc/markdown/pxf/using_pxf.html.md.erb [hunk 3]
```diff
@@ -88,6 +88,37 @@ To write data to an external data store with PXF, you create an external table w
...
+* `CHAR`, `TEXT`, array of `TEXT`
+* `DATE`, `TIMESTAMP`.
+
+PXF filter pushdown is supported for these operators:
+* `<`, `<=`, `>=`, `>`
+* `<>`, `=`
+* `IN`
+* `LIKE` (only for `TEXT` fields).
+
+To summarize, all of the following criteria must be met for filter pushdown to occur:
```
[the remark this answers is not in the document]
Comment (dyozie) [reply gpdb-doc/markdown/pxf/using_pxf.html.md.erb]: Agreed.  Still seems premature for this, as we don't have formal pushdown support just yet.

**Comment by dyozie** [comment]
@kongyew - please review whether this "experimental" wording for HBase/Hive pushdown is ok for 5.10.  Thanks!

File: gpdb-doc/markdown/pxf/hive_pxf.html.md.erb [hunk 4]
```diff
@@ -25,6 +25,8 @@ Apache Hive is a distributed data warehousing infrastructure. Hive facilitates m
 
 The PXF Hive connector reads data stored in a Hive table. This section describes how to use the PXF Hive connector. 
 
+**Note:** Filter pushdown with the Hive connector has not been tested, and should be considered an experimental feature in this release of Greenplum Database.
```
Comment (shivzone) [inline gpdb-doc/markdown/pxf/hive_pxf.html.md.erb]: Hive filter pushdown functionality has been completely verified and tested

File: gpdb-doc/markdown/pxf/using_pxf.html.md.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (shivzone) [reply gpdb-doc/markdown/pxf/using_pxf.html.md.erb]: Filter pushdown for Hive is certified as part of our CI.
We could leverage the information from here https://hdb.docs.pivotal.io/230/hawq/pxf/HivePXF.html#partitionfiltering

File: gpdb-doc/markdown/pxf/using_pxf.html.md.erb [hunk 5]
```diff
@@ -88,6 +88,37 @@ To write data to an external data store with PXF, you create an external table w
...
+
+You enable or disable filter pushdown for all external table protocols, including `pxf`, by setting the `gp_external_enable_filter_pushdown` configuration parameter. The default value is `off`; set it to `on` to enable filter pushdown.
+
+**Note:** Some data sources do not support filter pushdown. Also, filter pushdown may not be supported with certain data types or operators. If a query accesses a data source that does not support filter push-down for the query constraints, the query is instead executed without filter pushdown (the data is first transferred to Greenplum Database and then filtered). 
+
+PXF accesses data sources using different connectors, and filter pushdown support is determined by the specific connector implementation. 
+
+**Note:** Filter pushdown with the HBase and Hive connectors has not been tested, and should be considered an experimental feature in this release of Greenplum Database.
+
+PXF filter pushdown can be used with these data types:
```
Comment (lisakowen) [inline gpdb-doc/markdown/pxf/using_pxf.html.md.erb]: @shivzone - can you comment on the differences between the operators and data types supported that are identified here vs. those specifically mentioned in the hawq hiveORC discussion here: https://hdb.docs.pivotal.io/230/hawq/pxf/HivePXF.html#orc-profiles?  the operators mentioned on the hiveorc page do not seem to be a proper subset or superset of those mentioned here.

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 6]
```diff
@@ -3672,9 +3672,9 @@
   <topic id="gp_external_enable_filter_pushdown">
     <title>gp_external_enable_filter_pushdown</title>
     <body>
-      <p>Enable filter pushdown when reading data from external tables. If
-      pushdown fails, a query will be executed without it, and GPDB will apply
-      the same constraints to the result.</p>
+      <p>Enable filter pushdown when reading data from external tables. If pushdown fails, a query
+        is executed without pushing filters to the external data source (instead, Greenplum Database
+        applies the same constraints to the result).</p>
       <table id="enable_filter_pushdown_table">
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml]: Adding a link to "Defining External Tables" might be helpful

**Comment by dyozie** [comment]
@shivzone - I updated the docs to account for our new pushdown support in Hive/HBase.  Closing this PR for now.  Lisa will be opening a new one to port additional pushdown details from HAWQ.

**Pull request merged**